File: l-umi/Week_34/BOJ_11403_경로_찾기.py
def floyd_warshall(graph):
    N = len(graph)
    
    for k in range(N):
        for i in range(N):
            for j in range(N):
                if graph[i][k] and graph[k][j]:
                    graph[i][j] = 1
    
    return graph

##################################################################
N = int(input())
graph = [list(map(int, input().split())) for _ in range(N)]

result = floyd_warshall(graph)


##################################################################
for row in result:
    print(' '.join(map(str, row)))

File: yunva17/Week_34/BOJ_11403_경로찾기/BOJ_11403_경로찾기.js
let input = require("fs")
  .readFileSync(
    "c:/Algorithm_Study/yunva17/Week_34/BOJ_11403_경로찾기/input.txt"
  )
  // .readFileSync('/dev/stdin')
  .toString()
  .trim()
  .split("\n");

const N = Number(input[0]);
const graph = input.slice(1).map((v) => v.split(" ").map(Number));

for (let k = 0; k < N; k++) {
  for (let i = 0; i < N; i++) {
    for (let j = 0; j < N; j++) {
      if (graph[i][k] && graph[k][j]) {
        graph[i][j] = 1;
      }
    }
  }
}

console.log(graph.map((v) => v.join(" ")).join("\n"));
